packages/cluster-ui/src/statementsPage/statementsPage.tsx

```diff
@@ -227,6 +228,16 @@ export class StatementsPage extends React.Component<
       );
   };
 
+  calculateTotalWorkload = (statements: AggregateStatistics[]) => {
+    let totalWorkload = 0;
+    statements.forEach(
```

Comment:
This is an O(n) operation on each page reload if I'm not mistaken. Is there a way to cache results and figure out when we need to recompute (i.e. statement state changes, or a filter has changed)?

I don't think this is a big deal but there might be something simple we can do. If not, it's fine to add a comment. Feel free to argue against me if you think it doesn't matter that much to recompute or if I'm wrong.

Reply:
Initially I was going to create the value on the selector and it would only be updated when statements were updated (I would also need to recalculate on filter changes), but because we notice that crdb and ui are using different selectors, it would need to be updated on both, they need to be in sync, and so on, so the suggestion I got was to work only on the ui component and not touch on selectors, since they're considering merging those two.

When a filter is done, it needs to be recalculate anyway, the improvements is when the user refreshes the page, is that a common behaviour? users refreshing the page frequently? If they do, I can make the changes on both selectors, if that is not such a frequent case we can leave as is and I can create a issue to improve this on the future, when we have only one set of selectors

Reply:
Since the `statements` in this case are derived from filtered data and none of this is in the Redux store, the only thing I can think to do is memoize this function somehow. I think this could work if you use a [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) with the statements data as the key and the result of the calculation as the value.